src/components/RepoCard.astro

```diff
@@ -0,0 +1,157 @@
+---
+import { Icon } from "astro-icon"
+
+export type Props = {
+	title: string
+	sourceUrl: string
+	stackblitzUrl: string
+	codesandboxUrl: string
+	gitpodUrl: string
+	previewImage: string | null
+}
+
+const {
+	title,
+	sourceUrl,
+	stackblitzUrl,
+	codesandboxUrl,
+	gitpodUrl,
+	previewImage,
+} = Astro.props
+---
+
+<article class="relative border border-astro-gray-500 bg-astro-gray-600">
+	<div class="noise z-0"></div>
+	<div class="aspect-video bg-astro-gray-700">
+		{
+			!!previewImage && (
+				<img src={previewImage} alt="" class="object-cover s-full" />
+			)
+		}
+	</div>
+	<div class="flex flex-col gap-4 p-5">
+		<h3 class="heading-4">{title}</h3>
+		<p>
+			<a href={sourceUrl} class="inline-flex items-center gap-1">
+				<Icon name="github" size={16} />
+				<span
+					class="text-sm text-astro-gray-200 underline underline-offset-[3px]"
+					>View repo on GitHub</span
+				>
+			</a>
+		</p>
+		<hr class="border-astro-gray-500" />
+		<div
+			class="flex h-10 w-max min-w-0 max-w-full gap-px divide-x divide-astro-gray-600 rounded-full bg-blue-purple-gradient"
+		>
+			<a
+				href={stackblitzUrl}
+				target="_blank"
+				rel="noopener noreferrer"
+				class="flex items-center gap-2 rounded-l-full px-4 text-sm transition hover:backdrop-brightness-75"
+			>
+				<Icon name="stackblitz" size={16} /> Open in StackBlitz
+			</a>
+			<details class="relative z-10" data-card-options>
+				<summary
+					class="flex h-full cursor-pointer list-none items-center rounded-r-full px-2 transition hover:backdrop-brightness-75"
+					title="More options"
+				>
+					<Icon name="chevron-down" size={20} />
+				</summary>
+				<ul
+					class="absolute right-0 top-full mt-2 flex w-max flex-col rounded bg-white p-2 text-astro-gray-700 shadow-md"
+				>
+					<a
+						href={gitpodUrl}
+						class="flex flex-row items-center gap-2 rounded-sm px-3 py-2 hover:bg-blue-purple-gradient hover:text-white"
+					>
+						<Icon name="gitpod" size={20} /> Open in Gitpod
+					</a>
+					<a
+						href={codesandboxUrl}
+						class="flex flex-row items-center gap-2 rounded-sm px-3 py-1.5 hover:bg-blue-purple-gradient hover:text-white"
+					>
+						<Icon name="codesandbox" size={20} /> Open in CodeSandbox
+					</a>
+				</ul>
+			</details>
+		</div>
+	</div>
+</article>
+
+<script>
+	const dropdowns = document.querySelectorAll<HTMLDetailsElement>(
```

Comment:
Would have been so nice if Safari actually shipped the `class MyDetails extends HTMLDetailsElement` 😞